src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -246,84 +261,84 @@ public void resetNeededResources()
     }
 
     /**
-     * Check if the builder requires a certain ItemStack for the current construction.
-     *
-     * @param stack the stack to test.
-     * @return true if so.
+     * Update the available resources.
+     * which are needed for the build and in the builder's chest or inventory
      */
-    public boolean requiresResourceForBuilding(ItemStack stack)
+    private void updateAvailableResources()
     {
-        return neededResources.containsKey(stack.getUnlocalizedName());
-    }
+        final EntityCitizen builder = getWorkerEntity();
 
-    /**
-     * Provides a view of the builder building class.
-     */
-    public static class View extends AbstractBuildingWorker.View
-    {
-        private HashMap<String, Integer> neededResources;
-
-        /**
-         * Public constructor of the view, creates an instance of it.
-         *
-         * @param c the colony.
-         * @param l the position.
-         */
-        public View(final ColonyView c, final BlockPos l)
+        InventoryCitizen builderInventory = null;
+        if (builder!=null)
         {
-            super(c, l);
+            builderInventory = builder.getInventoryCitizen();
         }
 
-        /**
-         * Gets the blockOut Window.
-         *
-         * @return the window of the builder building.
-         */
-        @NotNull
-        @Override
-        public Window getWindow()
-        {
-            return new WindowHutBuilder(this);
-        }
 
-        @Override
-        public void deserialize(@NotNull ByteBuf buf)
+        for (@NotNull final Map.Entry<String, BuildingBuilderResource> entry : neededResources.entrySet())
         {
-            super.deserialize(buf);
+            BuildingBuilderResource resource = entry.getValue();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'resource' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)